artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQMessageProducer.java

```diff
@@ -577,26 +587,62 @@ private CompletionListenerWrapper(CompletionListener listener,
 
       @Override
       public void sendAcknowledged(org.apache.activemq.artemis.api.core.Message clientMessage) {
-         if (jmsMessage instanceof StreamMessage) {
-            try {
-               ((StreamMessage) jmsMessage).reset();
-            } catch (JMSException e) {
-               // HORNETQ-1209 XXX ignore?
+         if (active.get()) {
```

Comment:
Doing a getAndSet seems like it might be nicer given the aim to ensure only a single processing and call on the CompletionListener occurs. Granted though its not likely that multiple concurrent attempts are going to be made here.